okhttp/src/main/java/okhttp3/RealCall.java

```diff
@@ -33,7 +33,10 @@
 final class RealCall implements Call {
   final OkHttpClient client;
   final RetryAndFollowUpInterceptor retryAndFollowUpInterceptor;
-  final EventListener eventListener;
+
+  // There is a cycle between the Call and EventListener that makes this awkward. This will be set
+  // after we create the call instance then create the event listener instance.
+  volatile EventListener eventListener;
```

Comment:
We need to prevent OkHttp from dereferencing `eventListener` before it’s initialized. All of the ways we will access this field already have a happens-before relation; ie. the assignment will completes before`execute()` is called.